packages/babel-parser/src/types.ts

```diff
@@ -57,9 +57,11 @@ export interface NodeBase {
 
 // Using a union type for `Node` makes type-checking too slow.
 // Instead, add an index signature to allow a Node to be treated as anything.
-export type Node = NodeBase & {
+// todo(flow->ts): this probably should be replaced with union
+export interface Node extends NodeBase {
+  type: string;
   [key: string]: any;
-};
+}
```

Comment:
this is what is is causing most of "new type errors"(previous definition of Node was effectively same as `any`)